Fix bug when reading system.parts using UUID (issue 61220).

wudanzy opened the pull request:
During the predicates pushing down of system.parts table, there is a
logic to use that predicates to filter on the storages. This makes sense
if the predicate is on the `database`, `table`, `engine`. But it can be
problem if the predicate is about `uuid`, because storage UUID is
apparently different from parts UUID.

Rename the column name from `uuid` to `storage_uuid` fixed this.

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix incorrect results when filtering `system.parts` or `system.parts_columns` using UUID.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

Commit:
Fix bug when reading system.parts using UUID (issue 61220).

During the predicates pushing down of system.parts table, there is a
logic to use that predicates to filter on the storages. This makes sense
if the predicate is on the `database`, `table`, `engine`. But it can be
problem if the predicate is about `uuid`, because storage UUID is
apparently different from parts UUID.

Rename the column name from `uuid` to `storage_uuid` fixed this.

robot-ch-test-poll4 commented:
<!-- automatic status comment for PR #61479 from wudanzy/ClickHouse:61220_fix_read_system_parts -->
*This is an automated comment for commit adaf262dc09ff08628d54d5ffd64f7ea5053482d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61479/adaf262dc09ff08628d54d5ffd64f7ea5053482d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

Commit:
Add database = currentDatabase() condition to query test 03010_read_system_parts_table_test.sql

The failed test suggest to add database = currentDatabase() condition to tests that queries on system.parts.

antaljanosbenjamin commented on tests/queries/0_stateless/03010_read_system_parts_table_test.sql:
```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS users;
```
Could you please also add tests for `parts_columns`, `projection_parts` and `projection_parts_columns`? They all use the same base class, in theory they should behave the same, but better to have a test for them.

wudanzy replied:
Hi János, thanks for the suggestion! Just had a try, but found out that [`projection_parts`](https://github.com/ClickHouse/ClickHouse/blob/e82cf5d5ba62ec4e13288052e690228707e9de4e/src/Storages/System/StorageSystemProjectionParts.cpp#L21) and [`projection_parts_columns`](https://github.com/ClickHouse/ClickHouse/blob/e82cf5d5ba62ec4e13288052e690228707e9de4e/src/Storages/System/StorageSystemProjectionPartsColumns.cpp#L21) do not have a column called `uuid`. 

Added test for `projection_parts`.

antaljanosbenjamin reviewed (commented):
The changes looks good, but could you please merge/rebase master? I think the clang-tidy failures are fixed on master.

Commit:
Add tests for system.parts_columns table.

wudanzy committed:
Merge branch 'ClickHouse:master' into 61220_fix_read_system_parts

wudanzy commented:
Hi János, I am new to Clickhouse, could you please share a guidance on how to debug failing tests?

For Bugfix Validation that checks that either a new test (functional or integration) or there some changed tests that fail with the binary built on master branch, I am sure it did fail. https://fiddle.clickhouse.com/6891df0b-0a5d-42c7-a104-4a58d93799c1

antaljanosbenjamin commented:
Hi Dan!

For the bugfix validation the issue is not in your code. [This PR](https://github.com/ClickHouse/ClickHouse/pull/61363) will fix it. Let's wait a bit, hopefully it will get merged soon. Then you can merge a master to your branch so the updated config should run the bugfix check properly.

The RISC-V build and the single stateless test failure is also not related to your code.

antaljanosbenjamin commented:
Everything looks fine:
- build check failed on RISC-V which is a known issue
- bugfix failed because of a known issue, tested locally the the test only passes if I remove the filters on `uuid`, so the fix probably right
- the failed stateless test (`02481_async_insert_race_long`) is flaky
